src/root.cpp

```diff
@@ -59,17 +59,17 @@ Root::~Root()
     tags()->setMonitorManager({});
 
     // Note: delete in reverse order of initialization!
-    delete monitors();
-    delete clients();
+    monitors.reset();
```

Comment:
Aren't those redundant because it's going to happen in the destructor of `Child_` anyway (because `pointer` is now a `unique_ptr`?

I don't really mind offering `Child_::reset()`, but I'm wondering if there's ever an actual need for it.

That being said, I'm hugely in favor of dropping the `operator=` in `Child_` (as you did), and being more explicit in that it is a wrapper around `unique_ptr`.

Reply:
In root.cpp, the destruction order matters and this is why I'd like to make it explicit. I'm optimistic that we can get rid of this at some point.

Reply:
:+1: